query/src/storages/fuse/fuse_table.rs

```diff
@@ -172,27 +173,48 @@ impl Table for FuseTable {
 }
 
 impl FuseTable {
-    pub(crate) fn snapshot_loc(&self) -> Option<String> {
+    fn snapshot_loc(&self) -> Option<String> {
         self.table_info
             .options()
             .get(FUSE_OPT_KEY_SNAPSHOT_LOC)
             .cloned()
     }
 
+    pub fn snapshot_format_version(&self) -> Result<u64> {
+        Self::parse_snaphost_format_version(&self.table_info)
+    }
+
+    pub fn parse_snaphost_format_version(table_info: &TableInfo) -> Result<u64> {
+        table_info
+            .options()
+            .get(FUSE_OPT_KEY_SNAPSHOT_VER)
```

Comment:
Do we have the default version? If so, where the default defined?

Reply:
Yes, we should have a default value of snapshot version, which is 0, or `v0::Snapshot::VERSION`. let me fix it.

format-versions other than v0::Snapshot should have a table option explicitly defined in table options.

Reply:
We can set a current default version, upgrade it by hardcode if we changed the format, why need the table options to defined here?

Reply:
> Do we have the default version?

Some basic principles of this PR (should have been written in the PR's summary:- )   :

-  users other than `VersionedReader`  should always be working on the data structure exported by `meta::current`
 -  `io::meta_readers` always return the CURRENT version of the table metadata
    - conversion between versions is explicitly executed inside the reader.
    - backward compatibility if enforced by the implementation of `From <Old> for New`
      and of course, depends on that the converters are implemented correctly
 -  writers always write the table metadata of the CURRENT format-version
 - with the following exceptions
    - During the merge of TableSnaphost, versions of segments are kept (one snapshot may refer to different versions of segments)
    - The previous TableSnaphost of a given table snapshot may have a different format version.